osu.Game/Online/Leaderboards/TopLocalRank.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+using osu.Game.Online.API;
+using osu.Game.Rulesets;
+using osu.Game.Scoring;
+
+namespace osu.Game.Online.Leaderboards
+{
+    public class TopLocalRank : Container
+    {
+        private readonly BeatmapInfo beatmap;
+        private readonly UpdateableRank rank;
+
+        private ScoreManager scores;
+        private IBindable<RulesetInfo> ruleset;
+        private IAPIProvider api;
+
+        /// <summary>
+        /// Raised when the top score is loaded
+        /// </summary>
+        public Action<ScoreInfo> ScoreLoaded;
+
+        public TopLocalRank(BeatmapInfo beatmap)
+        {
+            this.beatmap = beatmap;
+
+            RelativeSizeAxes = Axes.Both;
+
+            InternalChild = rank = new UpdateableRank(null)
+            {
+                RelativeSizeAxes = Axes.Both
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ScoreManager scores, IBindable<RulesetInfo> ruleset, IAPIProvider api)
+        {
+            this.scores = scores;
+            this.ruleset = ruleset;
+            this.api = api;
+
+            FetchAndLoadTopScore();
+        }
+
+        public void FetchAndLoadTopScore()
+        {
+            var score = fetchTopScore();
+
+            loadTopScore(score);
+        }
+
+        private void loadTopScore(ScoreInfo score)
+        {
+            Schedule(() => rank.Rank = score?.Rank);
+
+            ScoreLoaded?.Invoke(score);
+        }
+
+        private ScoreInfo fetchTopScore()
+        {
+            if (scores == null || beatmap == null || ruleset?.Value == null || api?.LocalUser.Value == null)
+                return null;
+
+            return scores.GetAllUsableScores()
```

Comment:
Use `scores.Query()` instead.

Reply:
I changed this to `.QueryScores`. Is there a way to get the top score using only `.Query`? (`.OrderByDescending(s => s.TotalScore).FirstOrDefault()`)

Reply:
As you've done it is fine, point is to use a query. Previously it would retrieve/allocate all scores in the database and then match on them, whereas this translates into an SQL query that returns a few rows.